misc/python/materialize/cli/mkrelease.py

```diff
@@ -104,6 +125,23 @@ def main(version: str, checkout: Optional[str], create_branch: str, tag: bool) -
         say(f"Create a PR with your branch: '{branch}'")
 
 
+def update_versions_list(released_version: semver.VersionInfo) -> None:
+    """Update the doc config with the passed-in version"""
+    today = date.today().strftime("%d %B %Y")
+    toml_line = f'  {{ name = "v{released_version}", date = "{today}" }},\n'
+    with open(USER_DOC_CONFIG) as fh:
+        docs = fh.readlines()
+    wrote_line = False
+    with open(USER_DOC_CONFIG, "w") as fh:
+        for line in docs:
+            fh.write(line)
+            if line == "versions = [\n":
+                fh.write(toml_line)
+                wrote_line = True
+    if not wrote_line:
+        raise errors.MzRuntimeError("Couldn't determine where to insert new version")
```

Comment:
Did you consider using the `toml` library? You could do something like:

````
with open(USER_DOC_CONFIG) as fh:
    docs = toml.load(fh)

try:
    docs['params']['versions'].prepend({'name': release_version, 'date': today})
except KeyError:
    raise errors.MzRuntimeError("Couldn't determine where to insert new version")

with open(USER_DOC_CONFIG, "w") as fh:
    toml.dump(fh)
````

Reply:
Yes, but in addition to the toml dump format being radically different different, it also loses comments.

Also this isn't so hard to read or fragile that I think it's likely to be a large burden going forward.